main.py

```diff
@@ -0,0 +1,173 @@
+
+            filename = dump_to_file(daystring, events[daystring])
+            target_path = ""
+            if S3_PATH is not None:
+                target_path = dt.strftime(S3_PATH)
+            target_path += "/" + filename
+            upload_backup(s3, filename, target_path)
+            os.remove(filename)
+
+            # write number of entries per day to redis
+            redis.setex(daystring, redis_retention, str(len(events[daystring])))
+        except Exception as ex:
+            sys.stderr.write("ERROR: No backup created for %s\n" % daystring)
+            sys.stderr.write(traceback.format_exc() + "\n")
+
+    del redis
+    del s3
+    del events
+
+    print("Done for today.")
+
+
+def fetch_events():
+    has_more = True
+    params = {"limit": 100}
+    auth = HTTPBasicAuth(STRIPE_API_KEY, '')
+    while has_more:
+        r = requests.get("https://api.stripe.com/v1/events",
+                         params=params,
+                         auth=auth)
+        r.raise_for_status()
+        j = r.json()
+        has_more = j["has_more"]
+        params["starting_after"] = j["data"][-1]["id"]
+        for item in j["data"]:
+            yield item
+
+
+def dump_to_file(daystring, events):
+    """
+    Creates a backup file from a list of events
+    """
+    dump = ""
+    hasher = hashlib.sha1()
+    for item in events:
+        j = json.dumps(item, sort_keys=True) + "\n"
+        hasher.update(j)
+        dump += j
+    sha1hash = hasher.hexdigest()
+    filename = daystring
+
+    # write to file
+    with open(filename, 'wb') as dumpfile:
+        print("%s: Dumping to file %s" % (daystring, filename))
+        dumpfile.write(dump)
+    del dump
+
+    # encrypt file
+    new_filename = filename + "_" + sha1hash[0:6] + ".jsonl.aes-256-cbc"
+    call(["openssl", "enc", "-aes-256-cbc", "-base64",
+          "-pass", "env:FILE_ENCRYPTION_PASSWORD",
+          "-in", filename, "-out", new_filename])
+    os.remove(filename)
+
+    return new_filename
+
+
+def upload_backup(s3conn, local_path, target_path):
+    """
+    Uploads a backup file to S3
+    """
+    file_pointer = open(local_path, "rb")
+    s3conn.upload(target_path, file_pointer)
+    file_pointer.close()
+
+
+if __name__ == "__main__":
+
+    # check for the existence of environment variables
+    required_env_vars = ["STRIPE_API_KEY", "AWS_ACCESS_KEY_ID",
+                     "AWS_SECRET_ACCESS_KEY", "S3_ENDPOINT",
+                     "S3_BUCKET"]
+    missing_env_vars = []
+    for v in required_env_vars:
+        val = os.getenv(v)
+        if val is None or val == "":
+            missing_env_vars.append(v)
+    if len(missing_env_vars):
+        sys.stderr.write("One or more required environment variables are missing: ")
+        sys.stderr.write(", ".join(missing_env_vars) + "\n")
+        sys.exit(1)
+
+    while True:
+        archive_events()
+
+        # Wait for a day
+        time.sleep(60 * 60 * 24)
```

Comment:
The only problem I see is that this is uncontrollable if the service is restarted or moved from one host to the other. Each restart will trigger the download again. 

I understand that currently the easiest thing for you to run it on the old platform. Although I'd prefer to have our stuff running in k8s too. The other alternative is to use a systemd-timer with fleet directly. 

With releaseit it is as simple as this (most of it is defining the datastructure): https://github.com/giantswarm/releaseit/blob/master/base/cluster_backup.go

Releaseit also handles our credentials and contains the api, userd, companyd services.